v1/topdown/eval.go

```diff
@@ -447,6 +447,11 @@ func (e *eval) evalStep(iter evalIterator) error {
 		case *ast.Term:
 			// generateVar inlined here to avoid extra allocations in hot path
 			rterm := ast.VarTerm(e.fmtVarTerm())
+
+			if e.partial() {
+				e.inliningControl.PushDisable(rterm.Value, true)
```

Comment:
Here, we're pushing frames on top of the same stack used for the `with` kw. It should be safe, as we should have a consistent unroll sequence. This might however impact performance for large stacks, as membership checks involves traversing the stack until a match is found.
An alternative would be to have two separate data structures for `with` and presence checks for "naked" terms. This new inlining control could possibly even be a simple map instead of a stack, as `rterm` should always be assigned a unique var name(?).